inginious/frontend/courses.py

```diff
@@ -10,7 +10,11 @@
 
 from inginious.common.courses import Course
 from inginious.frontend.accessible_time import AccessibleTime
+from natsort import natsorted
 
+_all_tags_cache = {}
```

Comment:
Caches are accessed using ``self._name`` key, meaning that only the course-related value is required for for a given course. Why aren't such variables not class attributes ?

By the way, can tags be loaded in the class ``__init__`` method ?